pkg/operator/controller/externaldns/deployment.go

```diff
@@ -332,11 +335,37 @@ func (r *reconciler) updateExternalDNSDeployment(ctx context.Context, current, d
 // externalDNSDeploymentChanged evaluates whether or not a deployment update is necessary.
 // Returns a boolean if an update is necessary, and the deployment resource to update to.
 func externalDNSDeploymentChanged(current, expected *appsv1.Deployment) (bool, *appsv1.Deployment) {
+	changed := false
 	updated := current.DeepCopy()
-	annotationChanged := externalDNSAnnotationsChanged(current, expected, updated)
-	containersChanged := externalDNSContainersChanged(current, expected, updated)
-	volumesChanged := externalDNSVolumesChanged(current, expected, updated)
-	return annotationChanged || containersChanged || volumesChanged, updated
+
+	if expected.Spec.Replicas != nil {
```

Comment:
Do you have to cover the case where `expected.Spec.Replicas == nil` and `current.Spec.Replicas !=nil`?

Reply:
I try to not enforce more than needed to not spawn more fights between the operator and the other actors. If something is not set in the desired state then we don't really care about it and let it be defaulted or set by the user. BTW the replicas field is a good example, it defaults to 1 if not set, then if we'd try to enforce `nil` (given the desired is `nil`) the operator will start fighting for with the API server.

Reply:
I see that expected.Spec.Replicas is always hard-coded to 1, not defaulted, which explains it.